### ISSUE TITLE
Training on Multiple GPUs and Encountering NCCL Timeout Error

### ISSUE THREAD

**Issue by Malacca1211** [li-ronghui/LODGE#30]
Dear Author,

I encountered an issue while training your code on a multi-GPU server, where I received an NCCL timeout error. Specifically, the training process stopped at around 50 epochs on RTX 3090 server and at around 320 epochs on RTX 4090 server. When using a single GPU, the process was killed at around the same number of epochs. I have tried several potential solutions, but unfortunately, none of them resolved the issue.

I was wondering if you have encountered this problem before and, if so, how you managed to fix it.

Thank you for your assistance.

Best regards

**Comment by li-ronghui** [comment]
Thank you for your interest in Lodge and for your question!
I’m not entirely sure if your issue is caused by an out-of-memory error. If it is, you might try removing the following line from line 892 in the Local_Module.py file:
`self.training_step_outputs.append(loss)`

**Comment by YCCCCM** [comment]
I don't think the problem is the GPU, but the fact that as you train CPU memory gradually increases, causing the server to run out of CPU memory, not GPU memory. I'm training 100 rounds on a 256GB RAM server and I have to interrupt points or the server goes down.
